cg/meta/workflow/microsalt.py

```diff
@@ -59,6 +62,54 @@ def process(self) -> Process:
             )
         return self._process
 
+    def get_case_path(self, case_id: str) -> List[Path]:
+        case_obj: models.Family = self.status_db.family(case_id)
+        lims_project: str = self.get_project(case_obj.links[0].sample.internal_id)
+
+        case_path_list: List[Path] = [
+            Path(path)
+            for path in glob.glob(f"{self.root_dir}/results/{lims_project}_*", recursive=True)
+        ]
+
+        return case_path_list
+
+    def clean_run_dir(self, case_id: str, yes: bool, dry_run: bool = False):
+        """Remove workflow run directories for a MicroSALT case."""
+
+        self.verify_case_id_in_statusdb(case_id)
+        self.check_analysis_ongoing(case_id=case_id)
+
+        case_path_list: List[Path] = self.get_case_path(case_id=case_id)
+
+        if not case_path_list:
+            LOG.info(
+                f"There is no case paths for case {case_id}. Setting cleaned at to {datetime.now()}"
+            )
+            if not dry_run:
+                self.clean_analyses(case_id=case_id)
+            return EXIT_SUCCESS
+
+        if dry_run:
+            LOG.info(f"Would have deleted: {case_path_list}")
+            return EXIT_SUCCESS
+
+        for analysis_path in case_path_list:
+            if yes or click.confirm(
+                f"Are you sure you want to remove all files in {analysis_path}?"
+            ):
+                if analysis_path.is_symlink():
+                    LOG.warning(
+                        f"Will not automatically delete symlink: {analysis_path}, delete it manually",
+                    )
+                    return EXIT_FAIL
```

Comment:
This block is repeated twice make it a unit tested function for import and reuse - keeping it DRY

Reply:
@henrikstranneheim Just so I have understood you correctly:

This function is similar to clean_run_dir in analysis.oy but differs in how it verifies path existence and that in microsalt it loops through the case path list. Do you want me to break out the general stuff to another function and only overwrite the specific stuff that microsalt clean_run_dir performs?

Reply:
Yes exactly. Break out parts that has a specific function and is repeated to separate functions that can be reused .